### ISSUE TITLE
Support for auto rotate key in azurerm_mssql_server_transparent_data_encryption

### ISSUE THREAD

**Issue by johnfriedrich** [hashicorp/terraform-provider-azurerm#14171]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
sql Server transparent data encryption should offer a boolean to enable key auto rotation like in the Azure UI
![image](https://user-images.githubusercontent.com/4679374/141494229-af2ad394-6c21-4c14-a577-e3f97a723385.png)

### New or Affected Resource(s)

* azurerm_mssql_server_transparent_data_encryption

### Potential Terraform Configuration

```hcl
resource "azurerm_mssql_server_transparent_data_encryption" "encrypt" {
  server_id = azurerm_mssql_server.main.id
  key_vault_key_id = key.id
  auto_rotate_key = true
}
```

### References

* #0000

**Comment by aristosvo** [comment]
Hi @johnfriedrich! Thanks for your issue.

I tried to set it both in the Portal and via the API, but in both cases it wasn't set at all.

Did you have a working setup and what did it take to fixate that option?

**Comment by johnfriedrich** [comment]
Hello,
thanks for looking into it. Well I was reading documentation and found this setting, thought it could be useful for us but did not find it in azurerm terraform and created this ticket. (Did not try to actually set this setting by then)

I can confirm that setting it via the portal does not work. Getting api errors in chrome dev console.

**Comment by aristosvo** [comment]
@johnfriedrich Oops, seems like we're too early 🙊:

> Auto-Rotation feature for Azure SQL DB and MI hasn’t yet been announced publicly.
The Portal changes have been deployed which is why customers are being to see the “Auto-Rotate key” option.

> We are hoping to release this in public preview by end of CY.

**Comment by MarcinGrinberg** [comment]
@aristosvo I don't know if this is a good moment to implement this feature. From my side I see that checkbox vanished in UI. Via Azure CLI there is still an option of enabling auto-rotation which I successfully did. Also confirmed that rotation works and updates the key in TDE configuration It took around 60 minutes to update after a key rotation.

**Comment by MarcinGrinberg** [comment]
@aristosvo FYI feature just got into preview:
https://techcommunity.microsoft.com/t5/azure-sql-blog/automated-key-rotation-for-tde-byok-now-available-in-preview-for/ba-p/3607932

**Comment by aristosvo** [comment]
@MarcinGrinberg I'll take a look!
